pkg/client/unversioned/request.go

```diff
@@ -427,8 +427,36 @@ func (r *Request) VersionedParams(obj runtime.Object, convertor runtime.ObjectCo
 		return r
 	}
 	for k, v := range params {
-		for _, vv := range v {
-			r.setParam(k, vv)
+		for _, value := range v {
+			// TODO: Move it to setParam method, once we get rid of
+			// FieldSelectorParam & LabelSelectorParam methods.
+			if k == unversioned.LabelSelectorQueryParam(r.apiVersion) && value == "" {
+				// Don't set an empty selector for backward compatibility.
```

Comment:
I agree with the first part.

>
 One scheme is not guaranteed to hold
all conversions either - so there's an inside out problem here we may have
to solve in the future.

If that's the case, we will need to change this code too, because currently we only have one convertor here.

Sorry -  I probably don't see some consequence that you can see, so I'm trying to understand them (also note that I'm not opting for not changing it back to versioned - I'm just trying to understand what it would give us now...)